### PR TITLE
Align octane addon blueprint more with classic blueprint

### PR DESCRIPTION
**Pull request by Panman8201**
There are several differences with the addon blueprints between ["classic"](https://github.com/ember-cli/ember-cli/tree/v3.9.0/blueprints/addon) and [octane](https://github.com/ember-cli/ember-octane-blueprint/tree/v0.20.1/packages/@ember/octane-addon-blueprint). Bring these two more in-line with each other to reduce changes and improve compatibility between the two.

```
ember addon my-classic-addon
ember addon my-octane-addon -b @ember/octane-addon-blueprint
diff -rq -x 'node_modules' -x '.git' my-classic-addon my-octane-addon
```

Diff of changes that I haven't taken action on yet: https://gist.github.com/Panman8201/d7ad266b4c3d3a46baeaae6c503133f7

### Outstanding questions;
1. ~~`ember-try` has been removed from `package.json`, was this intentional?~~
2. ~~If `ember-try` is re-added, how should testing be setup to work with versions prior to Octane? `.travis.yml` and `config/ember-try.js`~~
3. ~~`ember-disable-prototype-extension` has been removed from `package.json`, was this intentional?~~
4. ~~`ember-cli-htmlbars` was moved from `devDependencies` to `dependencies` in `package.json`, was this intentional?~~
5. Are there any additional polyfills needing to be added to `package.json` to improve backwards compatibility?
6. ~~Should `tests/dummy/config/targets.js` still dynamically add `IE 11` for CI and production env?~~
7. ~~Can anyone help figure out the proper way to turn / rename `gitignore` and `npmignore` into dot-files? [Here](https://github.com/ember-cli/ember-cli/blob/v3.9.0/blueprints/addon/index.js#L165-L171) is how I _think_ it's done in the classic blueprint but I don't understand how all the `index.js` hooks are supposed to work.~~

### CODE REVIEW

**Comment by Panman8201** [comment]
> @NullVoxPopuli 
> 
> 2. Addons should still be able to work with pre-octane apps, I'd say the current release train that ember-try tests is good
> 5. I think that'd be up to the addon maintainer and what they want to support / how for back?

These two are intertwined IMO. How is ember-try going to pass on pre-octane versions when octane specific features will be used? I'm assuming polyfills will need to be added to support prior versions in both cases (testing and backwards compat).

> 6. Ember still officially supports IE11 (for now) :-\

To add a bit more to this one, currently the octane blueprint has a static list of targets that includes IE11. Whereas, the "classic" blueprint dynamically adds IE11 for CI and prod env. I wasn't sure which way was more appropriate now days.

https://gist.github.com/Panman8201/d7ad266b4c3d3a46baeaae6c503133f7#file-targets-js-diff

**Comment by chriskrycho** [comment]
(4) is *not* related to types so far as I understand, but I have no context here.

**Comment by NullVoxPopuli** [comment]
Some answers:
1. nope, the addon blueprint was originally from the module unification add blueprint, and I think that lagged behind as it was
2. Addons should still be able to work with pre-octane apps, I'd say the current release train that ember-try tests is good
3. idk. @rwjblue ?
4. I think this is related to types, @chriskrycho ?
5. I think that'd be up to the addon maintainer and what they want to support / how for back?
6. Ember still officially supports IE11 (for now) :-\
7. your hunch is correct!

Thanks for working on this!

**Comment by Panman8201** [comment]
Any more help here? Looks like a rainy weekend here so I could wrap this up. 😄

File: packages/@ember/octane-addon-blueprint/index.js [hunk 1]
```diff
@@ -33,4 +33,7 @@ module.exports = {
     });
 
   }
+
+  // TODO: Rename gitignore and npmignore to dot-files
```
Comment (NullVoxPopuli) [inline packages/@ember/octane-addon-blueprint/index.js]: todo? :)

Comment (Panman8201) [reply packages/@ember/octane-addon-blueprint/index.js]: Yep, that's one of the things that I need help with, which is why this PR is [WIP] yet.

More specifically, I'm not sure how to rename a file (to prepend a dot ".") nor which hook(s) that it should be done in. The [current/"classic" addon blueprint does this too](https://github.com/ember-cli/ember-cli/blob/v3.10.1/blueprints/addon/index.js#L162-L170) but I'm having a hard time boiling it down to just renaming a file (it does much more).

**Comment by Panman8201** [comment]
Alright, I think this is ready for full review. I poured through the current addon blueprint and did my best at figuring out the differences. My only remaining question would be other polyfills needed, such as one for `{{fn}}`.

**Comment by rwjblue** [comment]
Thank you @Panman8201!

**Comment by Panman8201** [comment]
Shoot! I was making some more tweaks based on another diff. I'll submit another PR. :)

**Comment by Panman8201** [comment]
That was my fault BTW, should have put it back to WIP but wasn't expecting a merge w/o some review comments. 😄

**Pull request merged**